### PR TITLE
add example how to use filterById

### PR DESCRIPTION
**Pull request by lplewa**
### Checklist

- [ ] Code compiles without errors locally
- [ ] All tests pass locally
- [ ] CI workflows execute properly

- [ ] CI workflows, not executed per PR (e.g. Nightly), execute properly 
- [ ] New tests added, especially if they will fail without my changes
- [ ] Added/extended example(s) to cover this functionality
- [ ] Extended the README/documentation
- [ ] All newly added source files have a license
- [ ] All newly added source files are referenced in CMake files
- [ ] Logger (with debug/info/... messages) is used

### CODE REVIEW

File: examples/memspace_numa/memspace_numa.c [hunk 1]
```diff
@@ -40,52 +40,89 @@ int createMemoryProvider(umf_memory_provider_handle_t *hProvider,
...
 error_memspace:
+    umfMemspaceDestroy(hMemspace);
     return ret;
 }
 
-int main(void) {
-    umf_memory_provider_handle_t hProvider = NULL;
-    umf_result_t ret;
+// Function to create a memory provider which allocates memory from the specified NUMA node
+// By using filter functions.
```
Comment (lukaszstolarczuk) [inline examples/memspace_numa/memspace_numa.c:63]: .

File: examples/memspace_numa/memspace_numa.c [hunk 2]
```diff
@@ -40,52 +40,89 @@ int createMemoryProvider(umf_memory_provider_handle_t *hProvider,
...
     }
 
-    // Create the memory provider that allocates memory from the specified NUMA node
-    // In this example, we allocate memory from the NUMA node 0
-    ret = createMemoryProvider(&hProvider, 0);
-    if (ret != UMF_RESULT_SUCCESS) {
+    // umfMemspaceHostAllGet() return immutable memspace, so we need to create a mutable copy
+    result = umfMemspaceClone(hostAll, &hMemspace);
+    if (result != UMF_RESULT_SUCCESS) {
+        fprintf(stderr, "umfMempolicyCreate failed.\n");
```
Comment (lukaszstolarczuk) [inline examples/memspace_numa/memspace_numa.c:80]: `umfMemspaceClone`

File: examples/memspace_numa/memspace_numa.c (hunk 1 above)
Comment (lukaszstolarczuk) [inline examples/memspace_numa/memspace_numa.c:63]: `functions` (plural)? you could be more descriptive, this function only uses the filterById and precisely filter it by 1 numa node;

perhaps you could mention other filter functions available in umf (a link to docs)?

Comment (lplewa) [reply examples/memspace_numa/memspace_numa.c]: done

Comment (lplewa) [reply examples/memspace_numa/memspace_numa.c]: i removed plural

File: examples/memspace_numa/memspace_numa.c [hunk 3]
```diff
@@ -18,8 +18,9 @@
 #include "utils_examples.h"
 
 // Function to create a memory provider which allocates memory from the specified NUMA node
-int createMemoryProvider(umf_memory_provider_handle_t *hProvider,
-                         unsigned numa) {
+// By using umfMemspaceCreateFromNumaArray
```
[the remark this answers is not in the document]
Comment (lplewa) [reply examples/memspace_numa/memspace_numa.c]: Done

File: examples/memspace_numa/memspace_numa.c (hunk 2 above)
Comment (lplewa) [reply examples/memspace_numa/memspace_numa.c]: Done

**Review by PatKamin — APPROVED** [review]

**Review by lukaszstolarczuk — APPROVED** [review]

**Pull request merged**